chutney/ui/src/app/modules/scenarios/components/search-list/scenarios.component.ts

```diff
@@ -93,15 +96,10 @@ export class ScenariosComponent implements OnInit, OnDestroy {
             );
         });
 
-        this.getScenarios()
-            .then(r => {
-                this.scenarios = r || [];
-                this.applyDefaultState();
-                this.applySavedState();
-                this.applyUriState();
-                this.applyFilters();
-            })
-            .catch(err => console.log(err));
+        this.fetchAndUpdateScenario()
+        this.scenarioUpdatesubscription = interval(3000).subscribe(() => {
```

Comment:
Please unsubscribe in destroy method.

Reply:
Looking at the **searchSub$** subject, i would call the complete method on it in destroy method too.